### PR TITLE
move_bradius_&_background_color_to_csdpopup

### PR DESCRIPTION
**Pull request by Feichtmeier**
This fixes the firefox border-radius bleed, by cutting the border-radius (other themes use this way to avoid firefox's problem with round edges, too :|) but keeps the border-radius for regular GTK context-menus
image_2018-05-13_13-06-48
So this closes https://github.com/ubuntu/gtk-communitheme/issues/431

But this also brings border radius back for non-csd menus on all 4 edges
like in mypaint
image_2018-05-13_13-07-17

@clobrano @godlyranchdressing @madsrh @CraigD is this okay for you? I personally feel the cut of borders look weird anyways. But I could imagine that some think the other way!

### CODE REVIEW

**Comment by didrocks** [comment]
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr439`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr439` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

**Comment by Feichtmeier** [comment]
peek 2018-05-13 15-37
 Also added a padding to avoid hover bleed on several apps and removed the border radius for hover. This bugs out on different places and my idea would be to remove it and add a padding @madsrh what do you think?

**Comment by didrocks** [comment]
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr439`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr439` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

**Comment by madsrh** [comment]
@Feichtmeier I'm still a bit AFK, but this looks good! I would of course prefer it without padding and without bleed ;) 

I'm just thinking out loud here, but when we can't have that, I'm actually in doubt if this suggestion is better than simply removing the rounded corners? They both seem like design comprises (although this one fits with the suru style).

I'm happy either way :D

**Comment by Feichtmeier** [comment]
@madsrh this is the current solution:
peek 2018-05-13 17-26

I think it looks a bit weird. You want to select the row and now the shape of the menu, don't you?
Cant and materia use the solution with the padding and without border-radius of rows, too:
peek 2018-05-13 17-28
Or google on their website context-menus:
peek 2018-05-13 17-30

**Comment by madsrh** [comment]
I was happy with the current solution 🙉 
But adding the padding has two advantages: It fixes the bleed issue + each menu item has **exactly** the same rectangle space on hover.

If you/we want to add the padding, I think you should add a few more pixels. The current padding could give the impression that this wasn't on purpose. 

You would love OSX @Feichtmeier 🤣 because [it has the rounded edges and padding](https://georgegarside.com/blog/macos/all-the-little-things-osx-el-capitan-hidden-features/safari-download-copy-address.png) and it looks fine to me, BUT when you show me the two suggestions - [1](https://user-images.githubusercontent.com/15329494/39968775-e10f4390-56d2-11e8-9946-657e62e14ea9.gif) [2](https://user-images.githubusercontent.com/15329494/39967880-c92f503a-56c3-11e8-931a-f5ecee502303.gif) - I would vote for our current design (the other one has some issue with the padding 😆 )

> But this also brings border radius back for non-csd menus on all 4 edges

👎 IMO the rounded top edges makes it look detached from the menubar.

I'm curious to hear what @clobrano and the others think about this 😄

**Comment by Feichtmeier** [comment]
Yes I was happy with the current solution too. I made this all sound a little bit complicated let me try again with a little structure:

1) there is a very small border bleed in the first row of the context menu in gtk apps:
image
2) @godlyranchdressing had some work so that the border radius on hover only applies to the first and last child of the menus and needed a app specific fix for this in _apps.scss, sadly eye of gnome (the picture viewer) has this problem in the last row now:
image

3) to avoid all this fuzz I would like to add the solution with the padding and without border-radius on hover of rows (keep the border-radius of the whole menu)

4) I have no idea weither we can keep 0 0 $small_radius $small_radius for non csd menus but I have a look or ask @clobrano @godlyranchdressing  for help =D

**Comment by clobrano** [comment]
I'm AFK as well :) however I too love OSX style :D 

Let's say I like current design as well, but it's fundamental to fix Firefox. That said, I like also the idea of the padding, so the problem would only be the pop up menus. I haven't had time to check, but let's try to find a way to fix them, but between Firefox issue and popup menu, Firefox looks more important I think

**Comment by Feichtmeier** [comment]
@clobrano if we find a solution it's important that we don't move the background and border-radius back to the menu,
.menu,
.context-menu 
because otherwise firefox has the border radius too -.- That god damn firefox

**Comment by didrocks** [comment]
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr439`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr439` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

**Comment by didrocks** [comment]
A new test snap version is available using: `snap refresh communitheme --channel=edge/gtk-communitheme-pr439`.
Further available updates will track that pull request then.

Switch back to stable or edge snap with `snap refresh communitheme --stable` or `snap refresh communitheme --edge` once you are done with it!
You may want to download as well `gtk-common-themes` snap from `edge/gtk-communitheme-pr439` channel to test your snaps with those changes.

Replace `refresh` with `install` if you haven't installed the snap yet.

**Comment by Feichtmeier** [comment]
Closing this after communication with @godlyranchdressing due to just bad ideas from my side!
Let's stick with the firefox issue for now.

Speaking of firefox: the issues won't end with the context-menu's border radius bleed
image

The new CSD option also has a border bleed at the top window edges...
image

And the greatest challenge, though not our challenge, remains at finding a solution for their way of theming the whole window! Because they are also applying the theme to the headerbar!!!
image
So when people switch themes everything looks quiet messy ...
peek 2018-05-14 19-34

Btw... the selected text.... they don't suck up the css part of textview... but from view/iconview! Meaning that we cant adapt the text selection in nautilus to a blue because otherwise icon view selection will also have the blue selection! 

At a last though it might be worth not to keep an eye for firefox theming at all. If they don't offer a proper GTK theme adoption I would not waste hours and hours of time to their mess like I and we already did =)
